app/helpers/foreman_leapp/template_helper.rb

```diff
@@ -3,20 +3,13 @@
 module ForemanLeapp
   module TemplateHelper
     def build_remediation_plan(remediation_ids, host)
-      entries = PreupgradeReportEntry.where(id: remediation_ids, host: host)
-                                     .where.not(detail: nil)
-                                     .pluck(:detail)
-      result = +''
-
-      entries.each do |entry|
-        entry['remediations']&.each do |remediation|
-          next unless remediation['type'] == 'command'
-
-          result << "#{remediation['context'].join(' ')}\n"
-        end
-      end
-
-      result
+      PreupgradeReportEntry.remediation_details(remediation_ids, host)
+                           .reject { |e| e['remediations'].nil? }
+                           .map { |e| e['remediations'] }
+                           .flatten
+                           .select { |r| r['type'] == 'command' }
+                           .map { |r| "#{r['context'].join(' ')}\n" }
+                           .join('')
```

Comment:
If you first `map` and `flatten`, you will get an array of remediations and then you can use just `compact` to throw away the `nil`s

Reply:
Also the last `join` doesn't need any arguments in this case. Alternatively, the second map could skip adding newlines and the final join could join with newlines

Reply:
Well best I can do is this:
```
      PreupgradeReportEntry.remediation_details(remediation_ids, host)
                           .map { |e| e['remediations'] }
                           .compact
                           .flatten
                           .select { |r| r['type'] == 'command' }
                           .map { |r| "#{r['context'].join(' ')}\n" }
                           .join
```

Is it okay?

Reply:
Swap `compact` and `flatten` and it will be good to go